setup/src/Magento/Setup/Validator/IpValidator.php

```diff
@@ -66,14 +66,32 @@ public function validateIps(array $ips, $noneAllowed)
      */
     private function filterIps(array $ips)
     {
-        foreach ($ips as $ip) {
-            if (filter_var($ip, FILTER_VALIDATE_IP)) {
-                $this->validIps[] = $ip;
-            } elseif ($ip == 'none') {
-                $this->none[] = $ip;
-            } else {
-                $this->invalidIps[] = $ip;
+        foreach ($ips as $range) {
+            if ($range === 'none') {
+                $this->none[] = $range;
+                continue;
             }
+
+            $cidr = 32;
+            $ip = $range;
+            if (strpos($range, '/') !== false) {
+                list($ip, $cidr) = explode('/', $range);
+            }
+
+            $ipValidator = new \Zend_Validate_Ip(['allowipv6' => true, 'allowipv4' => true]);
+            if (!$ipValidator->isValid($ip)) {
+                $this->invalidIps[] = $range;
+                continue;
+            }
+
+            $ipv4Validator = new \Zend_Validate_Ip(['allowipv6' => false, 'allowipv4' => true]);
```

Comment:
We must not add code using ZF1 classes. Looks like there is ZF2 equivalent https://framework.zend.com/manual/2.1/en/modules/zend.validator.ip.html